core/templates/dev/head/pages/exploration_editor/simple_editor_tab/simple_editor_sidebar_directive.html

```diff
@@ -12,13 +16,16 @@
 
     <li class="sidebar-item" ng-repeat="question in questionList.getQuestions() track by $index">
       <div id="<[ID_PREFIX + question.getId()]>" class="sidebar-item-header">
-        <i ng-click="question.toggleCollapse()" class="collapse-icon material-icons">
+        <i ng-click="question.toggleCollapse()" class="collapse-icon material-icons" ng-hide="sidebarModeService.isSidebarInEditMode()">
           <[isCollapsed(question) ? '&#xE5C7;' : '&#xE5C5;']>
         </i>
+        <i ng-click="deleteQuestion(question)" class="delete-icon material-icons" ng-hide="sidebarModeService.isSidebarInReadonlyMode()">
+          <['&#xE872;']>
+        </i>
         <strong ng-click="scrollToQuestion(question)">Question <[$index + 1]></strong>
       </div>
 
-      <ul ng-show="question.isCollapsed()">
+      <ul ng-show="question.isCollapsed() && sidebarModeService.isSidebarInReadonlyMode()">
```

Comment:
On giving a deep thought, I guess I should stick with ng-hide in this case. As the user will switch between modes and ng-if will be expensive because it will trigger recreating the DOM and the associated scope time and again. While ng-show will temporarily hide it and will easily show it without the overhead of recreating scope and DOM. 
I hope I am getting it right? @seanlip @kevinlee12